src/find_email_addresses.py

```diff
@@ -0,0 +1,63 @@
+import sys
+import argparse
+import validators 
+import requests
+import requests.exceptions
+from bs4 import BeautifulSoup 		# wow, not working with Python3.6, using Python3.5 
+import re
+from collections import deque
+email_pattern = re.compile("[\w\.-]+@[\w\.-]+")
+
+
+def findObjects(rootUrl):
+		processed_urls=set()	# unique items
+		emails=set()		# unique items
+		urls=deque([rootUrl])	# good FIFO
+		# obtain urls beyond root 
+		while len(urls):
+			urlnext=urls.popleft()
+			if not urlnext in processed_urls:
+				processed_urls.add(str(urlnext))
+				try:
+					response=requests.get(urlnext)
+					page=BeautifulSoup(response.text, 'html.parser')
+					
+					#print("Collecting emails from {}".format(urlnext))
+					emailSet=set(re.findall(email_pattern, str(page)))
+					emails=emails.union(emailSet)
+					# collect href references
+					for atag in page.find_all("a"):
```

Comment:
Also, this route of logic seems to be allowing processing for all hrefs, not just ones that are part of root domain (ex ``*.jana.com`) like the challenge requirements specify. What changes would you make to fix that?

Reply:
I used tldextract in the past to extract core subdomain (e.g jana.com) but it was one-time need few years ago, it was not perfect, probably there are better methods by now. Even  tldextract may have  raised to a new level. Once you have it, you add condition to parse only urls with matching core subdomain. Will save processing time for sure.